esmvalcore/preprocessor/_volume.py

```diff
@@ -275,6 +277,60 @@ def volume_statistics(cube, operator):
     return _create_cube_time(src_cube, result, times)
 
 
+def axis_statistics(cube, axis, operator):
+    """Perform statistics along a given axis.
+
+    Operates over an axis direction. If weights are required,
+    they are computed using the coordinate bounds.
+
+    Arguments
+    ---------
+    cube: iris.cube.Cube
+        Input cube.
+    axis: str
+        Direction over where to apply the operator. Possible values
+        are 'x', 'y', 'z', 't'.
+    operator: str
+        Statistics to perform. Available operators are:
+        'mean', 'median', 'std_dev', 'sum', 'variance',
+        'min', 'max', 'rms'.
+
+    Returns
+    -------
+    iris.cube.Cube
+        collapsed cube.
+    """
+    try:
+        coord = cube.coord(axis=axis)
+    except iris.exceptions.CoordinateNotFoundError as err:
+        raise ValueError(
+            'Axis {} not found in cube {}'.format(
+                axis,
+                cube.summary(shorten=True))) from err
+    coord_dims = cube.coord_dims(coord)
+    if len(coord_dims) > 1:
+        raise NotImplementedError(
+            'axis_statistics not implemented for '
+            'multidimensional coordinates.')
+    operation = get_iris_analysis_operation(operator)
+    weights = None
+    if operator_accept_weights(operator):
+        coord_dim = coord_dims[0]
+        expand = list(range(cube.ndim))
+        expand.remove(coord_dim)
+        bounds = coord.core_bounds()
+        weights = np.abs(bounds[..., 1] - bounds[..., 0])
+        weights = np.expand_dims(weights, expand)
+        weights = da.broadcast_to(weights, cube.shape)
```

Comment:
```suggestion
        weights = np.abs(
            coord.core_bounds()[..., 1] - coord.core_bounds()[..., 0]
        )
        weights = iris.util.broadcast_to_shape(weights, cube.shape, coord_dims)
```

I think this can be simplified with[ `iris.util.broadcast_to_shape`](https://scitools-iris.readthedocs.io/en/latest/generated/api/iris/util.html#iris.util.broadcast_to_shape).

Reply:
I never had performance problems with this, as it as doesn't copy the data but rather returns repeated views. I think it will work fine.